podman_compose.py

```diff
@@ -264,6 +264,14 @@ def rec_subs(value, subs_dict):
     do bash-like substitution in value and if list of dictionary do that recursively
     """
     if is_dict(value):
+        if 'environment' in value and is_dict(value['environment']):
+            # Load service's environment variables
+            subs_dict = subs_dict.copy()
+            svc_envs = {k: v for k, v in value['environment'].items() if k not in subs_dict}
+            subs_dict.update(svc_envs)
```

Comment:
I think it's enough to do a dict update two lines below. rec_subs won't delete any of the entries in `svc_envs`, so everything what this `update` does will be overwritten to lines below.

If I'm wrong, then the reason needs to be added as a comment because others will have the same question when reading this code.